core/src/main/scala/kafka/log/UnifiedLog.scala

```diff
@@ -433,6 +433,22 @@ class UnifiedLog(@volatile var logStartOffset: Long,
     }
   }
 
+  /**
```

Comment:
nit: Could we also update the scaladoc of `def updateHighWatermark(hw: Long)` to remove the part about the follower updating the HWM. This is no longer the case.